extensions/PhabBugz/lib/WebService.pm

```diff
@@ -0,0 +1,123 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::PhabBugz::WebService;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base qw(Bugzilla::WebService);
+
+use Bugzilla::Attachment;
+use Bugzilla::Bug;
+use Bugzilla::BugMail;
+use Bugzilla::Error;
+use Bugzilla::Group;
+use Bugzilla::User;
+use Bugzilla::Util qw(correct_urlbase detaint_natural);
+use Bugzilla::WebService::Constants;
+
+use Bugzilla::Extension::PhabBugz::Constants;
+use Bugzilla::Extension::PhabBugz::Util qw(
+  create_revision_attachment
+  create_private_revision_policy
+  edit_revision_policy
+  get_bug_role_phids
+  get_project_phid
+  get_revision_by_id
+  intersect
+  make_revision_public
+  request
+);
+
+use Data::Dumper;
+
+use constant PUBLIC_METHODS => qw(
+  revision
+);
+
+sub revision {
+    my ( $self, $params ) = @_;
+
+    unless ( defined $params->{revision}
+        && detaint_natural( $params->{revision} ) )
+    {
+        ThrowCodeError( 'param_required', { param => 'revision' } );
+    }
+
+    # Obtain more information about the revision from Phabricator
+    my $revision_id = $params->{revision};
+    my $revision    = get_revision_by_id($revision_id);
+
+    my $revision_phid  = $revision->{phid};
+    my $revision_title = $revision->{fields}{title} || 'Unknown Description';
+    my $bug_id         = $revision->{fields}{'bugzilla.bug-id'};
+
+    my $user = Bugzilla::User->new( { name => AUTOMATION_USER } );
+    Bugzilla->set_user($user);
```

Comment:
So this might be okay... but I want to check JSON RPC. I think it's possible to chain RPC methods there, and it isn't clear if it would be checking the login creds a second time.
That would mean I could execute other APIs as this user.

Also, shouldn't there be some form of protection on this?
What if Malory calls it a hundred thousand times with curl?